src/puppetlabs/puppetdb/http.clj

```diff
@@ -108,6 +108,28 @@
          (rr/charset "utf-8")
          (rr/status code))))
 
+(defn upload-file!
+  "Copies an octet-stream filetype upload to the local dir."
+  [dir {:keys [tempfile filename]}]
+  (let [local (io/file (str dir "/" filename))]
+    (io/copy tempfile local)
+    local))
+
+(defn tar-response
+  [body filename]
+  (-> body
+      rr/response
+      (rr/header "Content-Type" "application/octet-stream")
+      (rr/header "Content-Disposition" (str "attachment; filename=" filename))
+      (rr/charset "utf-8")
+      (rr/status status-ok)))
+
+(defn streamed-tar-response
+  [producer filename]
+  (tar-response
+   (rio/piped-input-stream producer)
+   filename))
```

Comment:
@senior Since you wrote the json streaming code, wanna take a look here to make sure I'm not crazy here?